src/Acuminator/Acuminator.Analyzers/StaticAnalysis/ChangesInPXCache/Walker.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Acuminator.Utilities.Common;
+using Acuminator.Utilities.Roslyn;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Acuminator.Analyzers.StaticAnalysis.ChangesInPXCache
+{
+	internal class Walker : NestedInvocationWalker
+	{
+		private static readonly ISet<string> MethodNames = new HashSet<string>(StringComparer.Ordinal)
```

Comment:
@dropsonic what about "Clear"/"Remove" methods of PXCache?
Also shouldn't we consider the direct modifications of "Inserted"/"Updated"/"Deleted" collections?

Reply:
See the item in JIRA - I've asked Ruslan about that.
`Inserted` / `Updated` / `Deleted` are not collections, they are properties which return `IEnumerable` from the inner private collection which is inaccessible for direct modifications.